oauthdoctor/diag/config_test.go

```diff
@@ -252,6 +255,55 @@ func TestPrint(t *testing.T) {
 	}
 }
 
+func TestReplaceConfig(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+
+	now := time.Now().Format("2006-01-02_")
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Errorf("Error getting current dir: %s", err)
+	}
+
+	tests := []struct {
+		desc   string
+		cfg    ConfigFile
+		backup string
+	}{
+		{
+			desc: "Config file and backup file exists",
+			cfg: ConfigFile{
+				Filepath: filepath.Join(dir, "testdata"),
+				Filename: "python_config2",
+				Lang:     "python",
+			},
+			backup: "diag/testdata/python_config2_" + now,
+		},
+	}
+
+	for _, test := range tests {
+		backup := test.cfg.ReplaceConfig(DevToken, "randomToken")
+
+		config := test.cfg.GetFilepath()
+		_, err := os.Stat(config)
+		if os.IsNotExist(err) {
+			t.Errorf("%s\nConfig file (%s) is missing", test.desc, config)
+		}
+
+		if !strings.Contains(backup, test.backup) {
+			t.Errorf("%s\nBackup config file (%s) is missing and expecting %s", test.desc, backup, test.backup)
+		}
+
+		// Cleaning up files
```

Comment:
You only have one test in the slice, but since it is a slice of test I assume you want to add more. Is the "Cleaning up file" specific to each sub-test in the slice? If so, then consider making copies of the files for each test. If os.Rename fails then the other tests in the slice will fail.

Reply:
I only have one test case for now. Are you suggesting to not use a slice?

Reply:
If you intend on adding other cases to this test in the future keep the slice. If you know you will only have one test, the no reason to use the slice. Having just one test show that you are sure that is all that is needed. Again, demonstrating best practice to the world. 

I think it would be best to not use the slice until you know you have another case.